pkg/commands/isrunning/isrunning_test.go

```diff
@@ -0,0 +1,136 @@
+				mockClient.EXPECT().GetClusterInfo(gomock.Any()).Return(nil, fakeErr)
+				// Act
+				_, gotErr := isRunning.AreAllInstancesRunning("")
+				// Assert
+				Expect(gotErr).To(HaveOccurred())
+			})
+		})
+
+		When("GetClusterDeployment fails", func() {
+			It("the GetClusterDeployment should receive the correct cluster_id and bubble up the error", func() {
+				// Arrange
+				mockClient.EXPECT().GetClusterInfo(gomock.Any()).Return(cluster, nil)
+				mockClient.EXPECT().GetClusterDeployment(gomock.Eq(cluster.ID())).Return(nil, fakeErr)
+				// Act
+				_, gotErr := isRunning.AreAllInstancesRunning("")
+				// Assert
+				Expect(gotErr).To(HaveOccurred())
+			})
+		})
+
+		When("ListStoppedInstances fails", func() {
+			It("the ListStoppedInstances should receive the correct InfraID and bubble the error", func() {
+				// Arrange
+				mockClient.EXPECT().GetClusterInfo(gomock.Any()).Return(cluster, nil)
+				mockClient.EXPECT().GetClusterDeployment(gomock.Eq(cluster.ID())).Return(&clusterDeployment, nil)
+
+				mockClient.EXPECT().ListStoppedInstances(gomock.Eq(infraID)).Return(nil, fakeErr)
+				// Act
+				_, gotErr := isRunning.AreAllInstancesRunning("")
+				// Assert
+				Expect(gotErr).To(HaveOccurred())
+			})
+		})
+
+		When("there were no stopped instances", func() {
+			It("should succeed and return that all instance are running", func() {
+				// Arrange
+				mockClient.EXPECT().GetClusterInfo(gomock.Any()).Return(cluster, nil)
+				mockClient.EXPECT().GetClusterDeployment(gomock.Eq(cluster.ID())).Return(&clusterDeployment, nil)
+				mockClient.EXPECT().ListStoppedInstances(gomock.Eq(infraID)).Return([]*ec2.Instance{}, nil)
+				// Act
+				got, gotErr := isRunning.AreAllInstancesRunning("")
+				// Assert
+				Expect(gotErr).ToNot(HaveOccurred())
+				Expect(got).To(BeTrue())
+			})
+		})
+
+		When("there was an error getting StopInstancesEvents", func() {
+			It("should succeed and return that all instance are running", func() {
+				// Arrange
+				mockClient.EXPECT().GetClusterInfo(gomock.Any()).Return(cluster, nil)
+				mockClient.EXPECT().GetClusterDeployment(gomock.Eq(cluster.ID())).Return(&clusterDeployment, nil)
+				mockClient.EXPECT().ListStoppedInstances(gomock.Eq(infraID)).Return([]*ec2.Instance{&instance}, nil)
+				mockClient.EXPECT().ListStopInstancesEvents().Return(nil, fakeErr)
+				// Act
+				_, gotErr := isRunning.AreAllInstancesRunning("")
+				// Assert
+				Expect(gotErr).To(HaveOccurred())
+			})
+		})
+
+		When("there user provided cannot stop the instance", func() {
+			It("should succeed are report that the not all instances are running", func() {
+				// Arrange
+				mockClient.EXPECT().GetClusterInfo(gomock.Any()).Return(cluster, nil)
+				mockClient.EXPECT().GetClusterDeployment(gomock.Eq(cluster.ID())).Return(&clusterDeployment, nil)
+				mockClient.EXPECT().ListStoppedInstances(gomock.Eq(infraID)).Return([]*ec2.Instance{&instance}, nil)
+				event := cloudtrail.Event{Username: aws.String("12345")}
```

Comment:
once I have another test with this event, I will move it into the `vars` section